enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisteredRewriter.scala

```diff
@@ -39,6 +39,22 @@ class RegisteredRewriter(tokenContext: TokenContext) {
     val newPipelineInfo = mutable.HashMap[LogicalPlan, PipelineInformation]()
     var rewrites = Map[LogicalPlan, LogicalPlan]()
     val rewritePlanWithRegisters = topDown(Rewriter.lift {
+      // TODO: This is a hack. The rewrite framework should do it for us, but it is not going into
+      // the expressions map. Will need to revisit, just not right now.
+      case p@Projection(source, expressions) =>
```

Comment:
Not sure what you mean here. Like the comment says - the case below should be enough already.